// aleppo/lib/shared/constants/constants.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var MILISECOND = 1;
var SECOND = 1 * 1000;
var MINUTE = 1000 * 60;
var HOUR = 3.6e+6;
var DAY = 8.64e+7;
var WEEK = 6.048e+8;
var MONTH = 2.628e+9;
var YEAR = 3.154e+10;

var MILISECOND_LABEL = 'milli';
var SECOND_LABEL = 'sec';
var MINUTE_LABEL = 'min';
var HOUR_LABEL = 'hr';
var DAY_LABEL = 'dy';
var WEEK_LABEL = 'wk';
var MONTH_LABEL = 'mo';
var YEAR_LABEL = 'yr';

var DELAY_VALUES = exports.DELAY_VALUES = [MILISECOND, // 0
SECOND, // 1
MINUTE, // 2
HOUR, // 3
DAY, // 4
WEEK, // 5
MONTH, // 6
YEAR];

var DELAY_LABELS = exports.DELAY_LABELS = [MILISECOND_LABEL, SECOND_LABEL, MINUTE_LABEL, HOUR_LABEL, DAY_LABEL, WEEK_LABEL, MONTH_LABEL, YEAR_LABEL];

var WEEKDAYS = exports.WEEKDAYS = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];

var WEEKDAYSABBR = exports.WEEKDAYSABBR = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

var MONTHS = exports.MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

var MONTHSABBR = exports.MONTHSABBR = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];

var MONTH_DAY_COUNT = exports.MONTH_DAY_COUNT = [31, 0, // February - 28 days in a common year and 29 days in leap years
31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
