// aleppo/lib/date/helper/generator/readableDate/general.js
'use strict';

var _constants = require('../../../../shared/constants');

var _countBtw = require('../../../countBtw');

var _countBtw2 = _interopRequireDefault(_countBtw);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 *
 * @param {date} timeStamp.
 */

// year
var fullYear = function fullYear(ts) {
  return new Date(ts).getUTCFullYear();
};
var shortYaer = function shortYaer(ts) {
  return parseInt(fullYear(ts).toString().slice(2), 10);
}; // year as shortcut

// month
var monthNum = function monthNum(ts) {
  return new Date(ts).getUTCMonth() + 1;
}; // January is 0.
var shortMonth = function shortMonth(ts) {
  return _constants.MONTHSABBR[monthNum(ts) - 1];
};
var fullMonth = function fullMonth(ts) {
  return _constants.MONTHS[monthNum(ts) - 1];
};

var weekNum = function weekNum(ts) {
  return (0, _countBtw2.default)('week', new Date(new Date().getUTCFullYear(), 0, 1).getTime(), new Date(ts).getTime()).weeks;
};

// day
var dayNum = function dayNum(ts) {
  return new Date(ts).getUTCDay();
};
var shortDay = function shortDay(ts) {
  return _constants.WEEKDAYSABBR[dayNum(ts)];
};
var fullDay = function fullDay(ts) {
  return _constants.WEEKDAYS[dayNum(ts)];
};

// time
var hour24 = function hour24(ts) {
  return new Date(ts).getUTCHours();
};

// credit : https://stackoverflow.com/questions/8888491/how-do-you-display-javascript-datetime-in-12-hour-am-pm-format
var hour12 = function hour12(ts) {
  var hours = new Date(ts).getUTCHours();
  var minutes = new Date(ts).getUTCMinutes();
  var amPM = hours >= 12 ? 'pm' : 'am';
  hours %= 12;
  hours = hours || 12; // the hour '0' should be '12'
  minutes = minutes < 10 ? '0' + minutes : minutes;
  return hours + ':' + minutes + ' ' + amPM;
};

var minutes = function minutes(ts) {
  return new Date(ts).getUTCMinutes();
};

var seconds = function seconds(ts) {
  return new Date(ts).getUTCSeconds();
};

// timeStamp
var timeStamp = function timeStamp(ts) {
  return new Date(ts).getTime();
};

// returns only local time without date.
var localTime = function localTime(ts) {
  return new Date(ts).toLocaleTimeString();
};

// returns only local time without date.
var localDate = function localDate(ts) {
  return new Date(ts).toLocaleDateString();
};

// return full local date.
var local = function local(ts) {
  return new Date(ts).toLocaleString();
};

var utc = function utc(ts) {
  return new Date(ts).toUTCString();
};

module.exports = {
  // year
  fullYear: fullYear,
  shortYaer: shortYaer,
  // month
  monthNum: monthNum,
  shortMonth: shortMonth,
  fullMonth: fullMonth,
  // week
  weekNum: weekNum,
  // day
  dayNum: dayNum,
  shortDay: shortDay,
  fullDay: fullDay,
  // time
  hour24: hour24,
  hour12: hour12,
  minutes: minutes,
  seconds: seconds,
  // timeStamp
  timeStamp: timeStamp,
  // local
  localTime: localTime,
  localDate: localDate,
  local: local,
  utc: utc
};
