// aleppo/lib/is/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _is = require('./is');

Object.defineProperty(exports, 'isUn', {
  enumerable: true,
  get: function get() {
    return _is.isUn;
  }
});
Object.defineProperty(exports, 'isNull', {
  enumerable: true,
  get: function get() {
    return _is.isNull;
  }
});
Object.defineProperty(exports, 'isValid', {
  enumerable: true,
  get: function get() {
    return _is.isValid;
  }
});
Object.defineProperty(exports, 'isObj', {
  enumerable: true,
  get: function get() {
    return _is.isObj;
  }
});
Object.defineProperty(exports, 'isBool', {
  enumerable: true,
  get: function get() {
    return _is.isBool;
  }
});
Object.defineProperty(exports, 'isNum', {
  enumerable: true,
  get: function get() {
    return _is.isNum;
  }
});
Object.defineProperty(exports, 'isStr', {
  enumerable: true,
  get: function get() {
    return _is.isStr;
  }
});
Object.defineProperty(exports, 'isFn', {
  enumerable: true,
  get: function get() {
    return _is.isFn;
  }
});
Object.defineProperty(exports, 'isSymb', {
  enumerable: true,
  get: function get() {
    return _is.isSymb;
  }
});
Object.defineProperty(exports, 'isArr', {
  enumerable: true,
  get: function get() {
    return _is.isArr;
  }
});
Object.defineProperty(exports, 'isZeroLength', {
  enumerable: true,
  get: function get() {
    return _is.isZeroLength;
  }
});
Object.defineProperty(exports, 'isStrEmpty', {
  enumerable: true,
  get: function get() {
    return _is.isStrEmpty;
  }
});
Object.defineProperty(exports, 'isArrEmpty', {
  enumerable: true,
  get: function get() {
    return _is.isArrEmpty;
  }
});
